SciLifeLab/scilifelab
# Develop - revamp statusdb connection and refactoring

- make MetricsParsers return data instead of updating passed object reference
- move generic database functions to **init** file
- in statusdb:
  - add views
  - make separate document representations of database entries. Resolves #84
  - factor out functions that work on documents or several connections from connection classes
- make conditional import of psutil
- delete obsolete CouchdbHandler class - connections now done by Connection classes 
- add update function to qc module for updating project sample names as well as project names
- refactor parsing of qc documents to take into account return values from parsers
- status reports 
  - add configuration section for database names
  - add option to provide ordered_million_reads on a sample by sample basis. Resolves #86
  - add project aliases and sample aliases
  - reformat output table
- tests
  - add test function for couchdb existance
  - add project summary data
  - add tests for verifying integrity of document upload to a couchdb database
  - rename numbered tests to comply with unit testing: NOTE: there still are interdependent tests lingering...
  - all statusdb tests now rely on locally installed couchdb
  - remove obsolete tests
  - add runtests.py so tests paths and imports work
- resolve issue #85
- add functionality for merged analyses (merging several sample runs into one)
- add BcbioRunController that is subclassed by ProjectController and ProductionController
  - new controller adds vcf summary function and hs_metrics
  - remove old BcbioRunController from project.py
- add application qc of samples despite not present in statusdb
- new dry function dry_makedir
